crypto/asn1/a_bitstr.c

```diff
@@ -161,7 +161,7 @@ int ASN1_BIT_STRING_set_bit(ASN1_BIT_STRING *a, int n, int value)
             return 1;         /* Don't need to set */
         c = OPENSSL_clear_realloc(a->data, a->length, w + 1);
         if (c == NULL) {
-            ERR_raise(ERR_LIB_ASN1, ERR_R_MALLOC_FAILURE);
+            ERR_raise(ERR_LIB_ASN1, ERR_R_CRYPTO_LIB);
```

Comment:
Raised errors following a memory allocation call should just be removed IMO.  The memory allocation call will report file and line number.  Subsequently reporting that it was LIB_ASN1 isn't helpful.  It's obviously ASN1 from the file name.

In this instance, we'd get an error on lines 162 **and** 164.  Polluting the error stack.

Reply:
Yeah, for the cases where there is an `OPENSSL_*alloc()` call immediately before the `ERR_raise(ERR_LIB_FOO, ERR_R_CRYPTO_LIB)` call with just the NULL check in between, I agree it will be better to drop the `ERR_raise()` call.

Though in the rather rare event of an malloc error it would not hurt to have the essentially duplicate error report (in particular since the size of the error queue should not be an issue here), 
it would look a little weird and the about 400 now superfluous `ERR_raise()` calls just clog the source code and increase the binary footprint.